provider/core/src/response.rs

```diff
@@ -72,7 +73,45 @@ pub struct DataPayload<M>
 where
     M: DataMarker,
 {
-    pub(crate) yoke: Yoke<M::Yokeable, Option<RcWrap<[u8]>>>,
+    pub(crate) yoke: Yoke<M::Yokeable, Option<Cart>>,
+}
+
+/// The type of the "cart" that is used by `DataPayload`.
+#[derive(Clone)]
+pub struct Cart(
+    #[cfg(feature = "sync")] yoke::erased::ErasedArcCart,
+    #[cfg(not(feature = "sync"))] yoke::erased::ErasedRcCart,
+);
+
+impl Deref for Cart {
+    type Target = dyn yoke::erased::ErasedDestructor;
+    fn deref(&self) -> &Self::Target {
+        &*self.0
+    }
+}
+// Safe because both Rc and Arc are CloneableCart
+unsafe impl stable_deref_trait::StableDeref for Cart {}
+// Safe because both Rc and Arc are StableDeref
+unsafe impl yoke::CloneableCart for Cart {}
```

Comment:
```suggestion
// Safe because both Rc and Arc are StableDeref, and our Deref impl delegates to that Deref impl.
unsafe impl stable_deref_trait::StableDeref for Cart {}
// Safe because both Rc and Arc are CloneableCart, and our Deref impl delegates to that Deref impl.
unsafe impl yoke::CloneableCart for Cart {}
```